src/socket-activate/socket-activate.c

```diff
@@ -489,14 +507,16 @@ static int run(int argc, char **argv) {
         for (;;) {
                 struct epoll_event event;
 
-                if (epoll_wait(epoll_fd, &event, 1, -1) < 0) {
-                        if (errno == EINTR)
-                                continue;
+                if (!arg_now) {
```

Comment:
Also here, `if (epoll_fd >= 0)`
Then, the assertion requested by @YHNdnzj becomes not necessary.